### DISCUSSION TITLE
Parcel breaks when trying to load an image

### DISCUSSION THREAD

**Discussion by morenoh149** [speechly/speechly#328] [Q&A] [answered]
Starting from the speechly browser example https://github.com/speechly/speechly/tree/main/examples/browser-client-example

if we try to load an image from javascript
```
// app.js
var DOM_img = document.createElement("img");
import { URL } from 'parcel';
DOM_img.src = new URL("./right-arrow.png");

document.body.appendChild(DOM_img);
```
parcel tries to install extra deps for handling images. https://parceljs.org/recipes/image/
```
🚨 Build failed.

@parcel/core: Failed to resolve '../native' from 
'./node_modules/@parcel/node-resolver-core/lib/index.js'

  /Users/harrymoreno/Programming/speechly-example/node_modules/@parcel/node-resolver-core/lib/index.js:25:23
    24 | var _Wrapper = _interopRequireDefault(require("./Wrapper"));
  > 25 | var _native = require("../native");
  >    |                       ^^^^^^^^^^^
    26 | function _interopRequireDefault(obj) { return obj && obj.__esMo

@parcel/resolver-default: Cannot load file './native-wasm-browser.js' in
 './node_modules/@parcel/node-resolver-core'.
```

Has anyone tried loading images from the speechly examples?

**Comment by thisismatu** [comment, marked as answer]
Hi @morenoh149 👋 

The problem here seems to be that you are trying to import `URL` module from the `parcel` package and use it for constructing the image URL. However this is not required, Parcels documentation say:

> You can reference non-JavaScript assets such as images or videos from a JavaScript file using the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) constructor. These are resolved relative to the module by using `import.meta.url` as the base URL parameter. The first argument must be a string literal to be recognized (not a variable or expression).
https://parceljs.org/languages/javascript/#url-dependencies

Following their example, loading images works just fine like so:

```
let img = document.createElement('img');
img.src = new URL('image.jpg', import.meta.url);
document.body.appendChild(img);
```

If you need more help with Parcel related questions, check out their [GitHub discussions page](https://github.com/parcel-bundler/parcel/discussions).

**Reply by morenoh149** [reply]
Thanks. I ended up changing my approach. Instead of generating a new img src and updating the dom, I added all of my project's images in the html and hid them using css
```
.img-class { display: none }
```
then, using javascript apply and remove another class that sets the display to block.

Related docs https://parceljs.org/languages/html/#images
